src/org/exist/util/function/Either.java

```diff
@@ -52,18 +53,43 @@ public final boolean isRight() {
     public final RightProjection<L, R> right() {
         return new RightProjection(this);
     }
-    
+
+    /**
+     * Catamorphism. Run the first given function if left,
```

Comment:
interesting, though I do not yet fully understand the purpose. Looks like a neat clean construct to me....